src/index.tsx

```diff
@@ -209,10 +211,20 @@ export default class ReactToPrint extends React.Component<IReactToPrintProps> {
     public render() {
         const {
             trigger,
+            onBeforeGetContent
         } = this.props;
 
         return React.cloneElement(trigger(), {
-            onClick: this.handlePrint,
+            onClick: () => {
+                const onBeforeGetContentPromise = onBeforeGetContent();
```

Comment:
Nit: please change the name from `onBeforeGetContentPromise` to `onBeforeGetContentOutput` since we don't actually know if it is a Promise yet